From: AMEET M. PARANJAPE <aparanja@redhat.com>
Date: Tue, 28 Apr 2009 19:08:23 -0500
Subject: [ppc64] set error_state to pci_channel_io_normal
Message-id: 49F79A77.20500@REDHAT.COM
O-Subject: Re: [PATCH RHEL5.3 BZ496872] powerpc/pseries: Set error_state to pci_channel_io_normal in eeh_report_reset()
Bugzilla: 496872
RH-Acked-by: David Howells <dhowells@redhat.com>

The previous patch did not take into account the patch that adds eeh_enable_irq(),
which changes the location of this change.  I have re-diffed the patch against -141.el
and re-submitted it here.

Brewbuild:
https://brewweb.devel.redhat.com/taskinfo?taskID=1780251

--
Ameet M. Paranjape
aparanja@redhat.com
IBM/Red Hat POWER Liason
IRC name: aparanja

diff --git a/arch/powerpc/platforms/pseries/eeh_driver.c b/arch/powerpc/platforms/pseries/eeh_driver.c
index 6a9183a..56096eb 100644
--- a/arch/powerpc/platforms/pseries/eeh_driver.c
+++ b/arch/powerpc/platforms/pseries/eeh_driver.c
@@ -104,8 +104,6 @@ static void eeh_enable_irq(struct pci_dev *dev)
 {
 	struct device_node *dn = pci_device_to_OF_node(dev);
 
-	dev->error_state = pci_channel_io_normal;
-
 	if ((PCI_DN(dn)->eeh_mode) & EEH_MODE_IRQ_DISABLED) {
 		PCI_DN(dn)->eeh_mode &= ~EEH_MODE_IRQ_DISABLED;
 		enable_irq(dev->irq);
@@ -181,6 +179,8 @@ static void eeh_report_reset(struct pci_dev *dev, void *userdata)
 	if (!driver)
 		return;
 
+	dev->error_state = pci_channel_io_normal;
+
 	eeh_enable_irq(dev);
 
 	if (!driver->err_handler ||
